### PR TITLE
[Feature:TAGrading] Solution/Ta Notes Panel

### PR DESCRIPTION
**Pull request by mukul-kmr-jha**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Currently, there is no solution /TA notes panel

### What is the new behavior?
Partially closes the issue #5648 
In this PR "Solution/Ta notes" panel is added and this lets instructors use shared notes and points for each of the rubric-components. 
- Lets you add solutions and notes which can be used as a reference by other instructors grading students
- Shows the last edit and the name of the author
- Each rubric component is connected to a corresponding solution box.

image

### CODE REVIEW

File: site/app/templates/grading/electronic/GradingPanelHeader.twig [hunk 1]
```diff
@@ -56,6 +57,15 @@
             {{ self.renderPanelPositionSelector('student_info') }}
             </span>
         {% endif %}
+        <span class="grade-panel" id="solution_ta_notes_btn">
```
Comment (shailpatels) [inline site/app/templates/grading/electronic/GradingPanelHeader.twig:60]: HTML CSS Id's should use dashes not underscores

https://submitty.org/developer/coding_style_guide/css

File: site/app/templates/grading/electronic/GradingPanelHeader.twig [hunk 2]
```diff
@@ -56,6 +57,15 @@
...
+            <button title="Show Solution and TA notes"
+                    class="invisible-btn key_to_click"
```
Comment (shailpatels) [inline site/app/templates/grading/electronic/GradingPanelHeader.twig:62]: you don't need the "key_to_click" class for HTML buttons, the `onclick/click` event will fire on mouse press or mouse enter

File: site/app/templates/grading/electronic/SolutionTaNotesPanel.twig [hunk 3]
```diff
@@ -0,0 +1,72 @@
+{% import _self as self %}
+<div id="solution_ta_notes" class="rubric_panel">
```
Comment (shailpatels) [inline site/app/templates/grading/electronic/SolutionTaNotesPanel.twig:2]: dashes not underscores, same for CSS classes as well

File: site/app/templates/grading/electronic/SolutionTaNotesPanel.twig [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    <div class="row">
+        <span class="grading_label">Solution/TA Notes</span>
+    </div>
+    {% for component in solution_components %}
+        {{ self.renderSolutionTaNotesCont(
```
Comment (shailpatels) [inline site/app/templates/grading/electronic/SolutionTaNotesPanel.twig:7]: what does the "Cont" in renderSolutionTaNotesCont mean/stand for ?

File: site/public/css/electronic.css [hunk 5]
```diff
@@ -474,6 +474,10 @@ progress::-webkit-progress-value {
     padding: 20px;
 }
 
+.rubric_panel {
```
Comment (shailpatels) [inline site/public/css/electronic.css:477]: dashes instead of underscores

File: site/public/js/solution-ta-notes.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
+function updateSolutionTaNotes(gradeable_id, component_id) {
+  let data = {
+    solution_text: $(`#textbox-solution-${component_id}`).val().trim(),
+    component_id,
```
Comment (shailpatels) [inline site/public/js/solution-ta-notes.js:4]: what is this `component_id` used for, should it have a value associated with it ?

**Comment by jportorreal00** [comment]
@mukul-kmr-jha I also think the hotkey settings page should be updated. When you hover over every button they state their purpose and then their hot key. For example "Show/Hide Grading Rubric (Press G)". I think it should say "Show/Hide Solutions and TA notes (Press __ )"

File: site/public/js/solution-ta-notes.js (hunk 6 above)
Comment (mukul-kmr-jha) [reply site/public/js/solution-ta-notes.js:4]: Its a parameter needed for updating the solution notes...It is associated with the value of `component_id`. Basically in JS objects if you have both key and value names exactly the same you can use this [shorthand syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Property_definitions). I believe we are using this syntax in some JS files already.

**Comment by mukul-kmr-jha** [comment]
There is some dependency on the underscore naming and renaming it to dashes would need a fair amount of refactor... and once all the open PRs for the Ta-grading page get merge I will create a PR for all these refactors. @shailpatels

**Comment by shailpatels** [comment]
> There is some dependency on the underscore naming and renaming it to dashes would need a fair amount of refactor... and once all the open PRs for the Ta-grading page get merge I will create a PR for all these refactors. @shailpatels 

Ok sounds good

**Comment by mukul-kmr-jha** [comment]
Hii @jportorreal00 , the textarea resize and the hidden label for panel issues are fixed. Hotkey is currently broken for the Ta-grading page... no hotkeys are working... also there is no hotkey for the notebook panel (recently added panel)... so I think I can create a smallish PR for hotkey fix after this one is merged, or should I just add the changes here itself? 
Thanks:)

**Comment by jportorreal00** [comment]
@mukul-kmr-jha if the changes are small then I think you can probably add the hotkey changes here so all the buttons are a but more similar. If it takes a bit more work then I think a newer PR should be fine. I do think the button color should be unique like the others:
Autograding: Red, 
Rubric: Green, 
Files: Blue, 
Student Info: Black, 
Peer Info: Purple
Discussion: Darker Blue,
Regrade: Orange.
Since there already two blue buttons maybe it should be a different unique color? Like pink, a different shade of purple or green, etc.

Other than that everything seems to be working really well! Really nice job :)

**Comment by mukul-kmr-jha** [comment]
Let's discuss it today with its second version also :)

**Comment by bmcutler** [comment]
I'll merge this one now, seems to work well.
Please continue with the UI suggestions & discussions in future PRs

**Pull request merged**